pkg/storage/internal/sync/chunker_baskets.go

```diff
@@ -0,0 +1,58 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/storage/queryopts"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/wdk"
+	"github.com/go-softwarelab/common/pkg/must"
+)
+
+type chunkerBaskets struct {
+	repo Repository
+}
+
+func newChunkerBaskets(repo Repository) *chunkerBaskets {
+	return &chunkerBaskets{
+		repo: repo,
+	}
+}
+
+func (c *chunkerBaskets) Name() string {
+	return "baskets"
+}
+
+func (c *chunkerBaskets) MaxDivider() uint64 {
+	return 1
+}
+
+func (c *chunkerBaskets) IsApplicable(requestedEntities OffsetsLookup) bool {
+	_, ok := requestedEntities[wdk.OutputBasketEntityName]
+	return ok
+}
+
+func (c *chunkerBaskets) Process(ctx context.Context, userID int, limit, relativeOffset uint64, offsetsLookup OffsetsLookup, since *time.Time, result *wdk.SyncChunk) (num uint64, err error) {
```

Comment:
I thought it would be easier to use Paging directly,
that's why I prepared a suggestion for that https://github.com/bsv-blockchain/go-wallet-toolbox/pull/268